src/main/java/security/JwtAuthorizationFilter.java

```diff
@@ -0,0 +1,45 @@
+package security;
+
+import io.jsonwebtoken.ExpiredJwtException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+public class JwtAuthorizationFilter extends OncePerRequestFilter {
+
+    private final JwtTokenProvider jwtTokenProvider;
+
+    public JwtAuthorizationFilter(JwtTokenProvider jwtTokenProvider) {
+        this.jwtTokenProvider = jwtTokenProvider;
+    }
+
+    @Override
+    protected void doFilterInternal(jakarta.servlet.http.HttpServletRequest request, jakarta.servlet.http.HttpServletResponse response, jakarta.servlet.FilterChain filterChain) throws jakarta.servlet.ServletException, IOException {
+        String token = extractToken((HttpServletRequest) request);
+        if (token != null && jwtTokenProvider.validateToken(token)){
+            try {
+                String username = jwtTokenProvider.getUsernameFromToken(token);
+                Authentication auth = new UsernamePasswordAuthenticationToken(username, null, null);
+                SecurityContextHolder.getContext().setAuthentication(auth);
+            } catch (ExpiredJwtException ex) {
+            }
```

Comment:
inside the catch block, you should reject a user from being able to proceed with calling your API